src/factory.ts

```diff
@@ -1,5 +1,6 @@
 import { ethers, Contract, Signer } from "ethers";
 import { JsonRpcProvider } from "@ethersproject/providers";
+import { keccak256, solidityKeccak256, solidityPack } from "ethers/lib/utils";
```

Comment:
wouldn'T it be better to import `utils` from `ethers` and use it. I am not a js expert, so not 100% sure

Reply:
IIRC, there's no difference regarding performance, but I felt that the syntaxis is more readable if we omit the `utils.` - Can add it if you prefer it that way :-D

Reply:
Although this is just a minor nit and entirely up to you how you want to do it, when I think about how I intuitively make the decision here its about how many times it is used. So for example if I had used keccak256 100 times, it would start to look pretty redundant if it was always `ethers.utils.keccak256`. In this file it seems like these things are only used once or twice so I might stick with `ethers.utils.sweetFunction` here. Also you already have ethers imported so this entire import line could be gone!

Note that there is some inconsistent usage below -- I have commented there.

Reply:
@bh2smith you're totally right - will stick to ethers.utils.sweetFunction

Thanks for catching that 🙏